lib/foreman_tasks/tasks/export_tasks.rake

```diff
@@ -255,6 +257,16 @@ namespace :foreman_tasks do
         tasks.each_with_index do |task, count|
           File.open(File.join(tmp_dir, "#{task.id}.html"), 'w') { |file| file.write(PageHelper.pagify(renderer.render_task(task))) }
           puts "#{count + 1}/#{total}"
+        rescue StandardError => e
+          puts "WARNING: task failed to export. Additional task details below."
+          puts task.inspect
+          if skip_errors?
+            # Displaying task.id may result in another failure in case the task is severely broken (as seen on the field).
+            puts "WARNING: Skipping task #{count + 1} because it failed to export."
+          else
+            puts "Re-run with SKIP_FAILED=true if you want to simply skip task #{count + 1} and any other tasks that fail to export."
+            raise e
+          end
         end
 
         File.open(File.join(tmp_dir, 'index.html'), 'w') { |file| file.write(PageHelper.pagify(PageHelper.generate_index(tasks))) }
```

Comment:
Even if we skip the failed tasks, they appear in the index generated here, but the index may link to files which are not there because they failed to export

Reply:
And possibly if even `task.id` would raise an exception as suggested here https://github.com/theforeman/foreman-tasks/pull/588/files#diff-8898a6eafc1346188e475c0a876c1aa6R264 then the index generation would fail

Reply:
This is resolved by creating an exported_tasks array that is appended only the tasks that are successfully exported, then passing generate_index() this new array instead of the original one.
I expect this to fix the issue of generate_index() failing.